src/main/kotlin/cacophony/regex/RegexParser.kt

```diff
@@ -0,0 +1,149 @@
+package cacophony.regex
+
+import cacophony.utils.AlgebraicRegex
+import java.util.EmptyStackException
+import java.util.Stack
+
+fun parseRegex(str: String): AlgebraicRegex {
```

Comment:
im wondering if this could be a static mehod `AlgebraicRegex.fromString(str: String)`

apparently in Kotlin you do such things with [companion objects](https://kotlinlang.org/docs/object-declarations.html#companion-objects)

Reply:
I had same idea and found out about companion objects, however I wanted to avoid touching AlgebraicRegex class/file for the time being to not mess with concurrent work. But eventually, I agree

Reply:
I think you can freely move it, as long as you don't modify existing part, it won't cause any conflicts, and even if you do, it's part of the job :x

Reply:
You can declare an [extension function](https://kotlinlang.org/docs/extensions.html#extension-functions) to get this syntax without modifying `AlgebraicRegex`

Reply:
Added `AlgebraicRegex.fromString(str: String)` function